modules/encrypt/classes/Kohana/Encrypt/Engine/Openssl.php

```diff
@@ -138,36 +98,18 @@ public function decrypt(string $data)
         // If the payload is not valid JSON or does not have the proper keys set we will
         // assume it is invalid and bail out of the routine since we will not be able
         // to decrypt the given value. We'll also check the MAC for this encryption.
-        if (!$this->valid_payload($data))
-        {
-            // Decryption failed
-            return NULL;
-        }
-
-        if (!$this->valid_mac($data))
+        if (!$this->valid_payload($data) || !$this->valid_mac($data) || !$iv = base64_decode($data['iv']))
```

Comment:
Please let's not use same line assign and comparison. Assign $iv one line before if statement, thank you.